### PR TITLE
Avatar update

### PR DESCRIPTION
**Pull request by aveline**
### WHY are these changes introduced?

Fixes https://github.com/Shopify/polaris-summer-editions/issues/738

### WHAT is this pull request doing?

- Replaces circle shape with squircles
- Updates customer icon 

Screenshot 2023-06-29 at 12 14 56 PM

### How to 🎩

- Review in Storybook with beta flag turned on

### CODE REVIEW

File: polaris-react/src/components/Avatar/Avatar.tsx [hunk 1]
```diff
@@ -153,12 +168,33 @@ export function Avatar({
   // Use `dominant-baseline: central` instead of `dy` when Edge supports it.
   const verticalOffset = '0.35em';
 
-  const avatarBody =
-    customer || !initials ? (
+  const avatarPath = polarisSummerEditions2023 ? (
+    <>
```
Comment (sam-b-rose) [inline polaris-react/src/components/Avatar/Avatar.tsx:169]: Looks great! Only suggestion would be to maybe move these out of the component so there is less clutter inside the function logic. Not blocking though 👍 

```
const avatarPath = polarisSummerEditions2023
  ? avatarPathNext
  : avatarPathLegacy

// outside component
const avatarPathNext = <path ...
const avatarPathLegacy = <path ...
```

**Review by sam-b-rose — APPROVED** [review]
🚀

File: polaris-react/src/components/Avatar/Avatar.tsx (hunk 1 above)
Comment (aveline) [reply polaris-react/src/components/Avatar/Avatar.tsx:169]: The se23 hook needs to be inside the function though?

**Comment by sarahill** [comment]
Overall looks good but at the two smallest sizes that icon looks really thin when it gets scaled. Is there any way around that? 
cc: @zecarlostorre 

actual
image

expected
image

**Review by aaronccasanova — APPROVED** [review]

**Comment by aveline** [comment]
> Overall looks good but at the two smallest sizes that icon looks really thin when it gets scaled. Is there any way around that? cc: @zecarlostorre
> 
> actual image
> 
> expected image

The two smallest sizes have a stroke width of 1.5 as per [these instructions](https://github.com/Shopify/polaris-summer-editions/issues/738#issuecomment-1612692358), but we can easily remove that if it's not what we want.

**Comment by zecarlostorre** [comment]
I see the same problem @sarahill mentioned above.

I looked at storybooks and it looks like the 20x20 avatar is a scaled down 40x40 icon with 1.5 stroke, so it end up with a stroke that is half of that. (if I'm interpreting this right 😅 )

 Screenshot 2023-07-04 at 17 00 51
 
In any case, looking at the screenshot zoomed in, it being 1.5px I would expect some anti-aliasing at the bottom line, which we don't have.
Screenshot 2023-07-04 at 17 04 57

**Comment by aveline** [comment]
Yes the stroke gets scaled down (or up) as well. If we compensate for the stroke scaling and try to keep it closer to 2px or 1.5px depending on size, this is what we end up with. Is this closer to what we are expecting?
Screenshot 2023-07-04 at 9 38 28 AM
Screenshot 2023-07-04 at 9 38 46 AM

```  extraSmall: '3',
  small: '2.5',
  medium: '2.5',
  large: '2.5',
  'xl-experimental': '2',
  '2xl-experimental': '1.5',
  ```

**Comment by zecarlostorre** [comment]
@aveline yes! that looks good to me.
@sarahill what do you think?

**Review by sarahill — APPROVED** [review]
Looks great! 👏

File: polaris-react/src/components/Avatar/Avatar.tsx (hunk 1 above)
Comment (sam-b-rose) [reply polaris-react/src/components/Avatar/Avatar.tsx:169]: > The se23 hook needs to be inside the function though?

Yes, sorry the code above wasn't very clear 😅 

```jsx
export const Avatar = (props) => {
  //other code

  const avatarPath = polarisSummerEditions2023
    ? avatarPathNext
    : avatarPathLegacy

  // other code
}

/* Outside component */
const avatarPathNext = <path ...
const avatarPathLegacy = <path ...

Comment (aveline) [reply polaris-react/src/components/Avatar/Avatar.tsx:169]: Oh got it. The path also relies on the avatar size though which gets passed into the component. Will leave as is for simplicity.

**Pull request merged**